### PR TITLE
CORE-14685: Extend DSL to support changing virtual node status.

### PR DESCRIPTION
**Pull request by chrisr3**
Extend the driver's simulated network and DSL so that virtual nodes can be `SUSPENDED` or `PENDING` as well as `ACTIVE`.

### CODE REVIEW

File: testing/driver/driver-engine/src/main/kotlin/net/corda/testing/driver/sandbox/Constants.kt [hunk 1]
```diff
@@ -22,3 +22,18 @@ const val DEFAULT_KEY_SCHEME = ECDSA_SECP256R1_CODE_NAME
...
+    val iterator = iterator()
+    while (iterator.hasNext()) {
+        val item = iterator.next()
+        if (predicate(item)) {
+            destination += item
+            iterator.remove()
+        }
+    }
+    return destination
+}
```
Comment (yarivus) [inline testing/driver/driver-engine/src/main/kotlin/net/corda/testing/driver/sandbox/Constants.kt]: This aux function doesn't seem related to the Constants.kt file ... maybe it should bne moved to another file ?

**Review by yarivus — COMMENTED** [review]
See comment

File: testing/driver/driver-engine/src/main/kotlin/net/corda/testing/driver/sandbox/Constants.kt (hunk 1 above)
Comment (chrisr3) [reply testing/driver/driver-engine/src/main/kotlin/net/corda/testing/driver/sandbox/Constants.kt]: Maybe I could also rename the file to something more inclusive than "Constants"? Although personally speaking, I _hate_ "Utils"-like classes because they just end up becoming "buckets of stuff".

Need to think about this... :thinking:

**Review by yarivus — COMMENTED** [review]
thx @chrisr3
